crawl-ref/source/describe.cc

```diff
@@ -642,6 +642,34 @@ static string _randart_descrip(const item_def &item)
 }
 #undef known_proprt
 
+// If item is an unrandart with a DESCRIP field, return its contents.
+// Otherwise, return "".
+static string _unrandart_descrip(const item_def &item)
+{
+    if (!is_unrandom_artefact(item)) return "";
+    auto entry = get_unrand_entry(item.unrand_idx);
+    if (!entry->descrip) return "";
+    return string("\n")+entry->descrip;
```

Comment:
I'm concerned that you add a newline here and also at lines 1410, 1853, and 1960. (I haven't downloaded and built this for testing yet). The function `_artefact_descrip` shouldn't handle any newline padding before or after the description text, this should be left to the callers.

Reply:
_artefact_descrip() doesn't add any padding. There is a newline, but that's to ensure that the text appears at the start of a line. You get this throughout get_item_description().

1410, 1853 and 1960 do add a blank line, but this was already there.

Reply:
The script doesn't let you repeat lines in an artefact definition at the moment. Rather than change this, would something like this be acceptable?

DESCRIP: It enhances your poison magic.
+ It grants immunity to poison.
+ It can be evoked to poison those around you for a time.
+ It may poison the flesh of those it strikes, and is more effective with a high
 Evocations skill.

i.e. make + continue the previous field, but with a \n at the start? Thanks.

Reply:
> _artefact_descrip() doesn't add any padding. There is a newline, but that's to ensure that the text appears at the start of a line. You get this throughout get_item_description().

Right; but `_unrand_descrip()` (the line that I'm commenting on) adds this leading newline, while `_randart_descrip()` does not. The two functions should be consistent, and `_artefact_descrip()` should consistently return a string that starts with the first letter of the descrip text, not a newline.

Reply:
I'm a bit confused. _randart_descrip() either returns "", or returns a string which starts with a \n. The parts of the code which add text to the string it creates are:

```
const char* type = _jewellery_base_ability_description(item.sub_type);
description += "\n";
description += type;
```
...
```
sdesc = (string based on something in propdescs[])
description += '\n';
description += sdesc;

```
...

```
snprintf(buf, sizeof buf, "\nIt makes you %s%s stealthy.", PARAMS)
description += buf;

```
_artefact_descrip() can return several lines of text, which would need the correct number of \n characters between them. While it's possible for the function to strip the initial \n before it returns the string, the caller would only have to add it back in.

I've made a few more changes to artefact_descrip() now (mainly to eliminate the need for \n in art-desc.txt), but it still returns a string which starts with \n.